cpp/tests/test_math_time_series_functor.cpp

```diff
@@ -0,0 +1,118 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: René Schmieding
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "memilio/math/time_series_functor.h"
+#include "random_number_test.h"
+
+#include "gtest/gtest.h"
+#include <vector>
+
+using TestMathTimeSeriesFunctor = RandomNumberTest;
+
+TEST_F(TestMathTimeSeriesFunctor, zero)
+{
+    // Test that the default initialized functor always returns zero, using a random evaluation point.
+
+    const int num_evals = 100;
+
+    // initialize functor using the default ctor
+    mio::TimeSeriesFunctor<double> tsf;
+
+    // check one deterministic value first to avoid flooding the test output with failed tests
+    ASSERT_EQ(tsf(0.0), 0.0);
+    // verify output
+    for (int i = 0; i < num_evals; i++) {
+        auto random_t_eval = this->random_number();
+        EXPECT_EQ(tsf(random_t_eval), 0.0);
+    }
+}
+
+TEST_F(TestMathTimeSeriesFunctor, linearInterpolation)
+{
+    // Test that linear interpolation works for a piecewise linear function.
+
+    // continuous function that is constant 1 for t<0, linear in [0, 1] with slope 2, and constant 3 for t>1
+    const auto pcw_lin_fct = [&](double t) {
+        return 1 + 2 * std::clamp(t, 0.0, 1.0); // .../```
```

Comment:
What is the purpose of this comment?

Reply:
It is a little ascii art that shows how the function looks. I can remove it, if it is confusing.